### PR TITLE
[Addon] Update to Flux2

### LINKED ISSUE #705: Use of flux v2
Hi guys ✋ 

I wonder why the `fluxcd` addon is not using the latest version of the fluxcd such as v2. [Here](https://github.com/fluxcd/flux2), you can access it. 

Is there any plan to upgrade the addon regarding the next version?

### PR DESCRIPTION
**Pull request by kichristensen**
### Description of your changes

Updates the FluxCD addon to Flux2.

Fixes #705 

### How has this code been tested?

The addon has been tested in my local Kubevela instance.

### Checklist

I have:

- [X] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [X] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](https://github.com/kubevela/catalog/tree/master/examples).
- [ ] New addon should be put in [experimental](https://github.com/kubevela/catalog/tree/master/experimental/addons).
- [X] Update addon should modify the `version` in `metadata.yaml` to generate a new version.

### CODE REVIEW

**Comment by shlevy** [comment]
Anything remaining to do to get this merged?

File: addons/fluxcd/definitions/helm-release-def.cue [hunk 1]
```diff
@@ -61,23 +61,36 @@ template: {
...
 				metadata: {
 					name: context.name
 				}
 				if parameter.repoType == "git" {
-					kind: "GitRepository"
+					apiVersion: "source.toolkit.fluxcd.io/v1"
+					kind:       "GitRepository"
 					spec: {
 						url: parameter.url
 						if parameter.git.branch != _|_ {
```
Comment (FogDong) [inline addons/fluxcd/definitions/helm-release-def.cue:72]:
```suggestion
if parameter.git != _|_ {
  ref: parameter.git
}
```

**Comment by FogDong** [comment]
Please resolve the conflicts.

**Comment by Gideon-Felt** [comment]
@kichristensen Has progress on this stalled?

**Comment by andrea-berling** [comment]
@FogDong can you check the MR I opened with the conflicts resolved? #756 
@Gideon-Felt please upvote the MR I opened, the more interest on it the higher the chance this gets merged and Flux gets upgraded (I hope)

**Pull request merged**